Gems/RecastNavigation/Code/Tests/EditorNavigationMeshTest.cpp

```diff
@@ -0,0 +1,472 @@
+    using AZ::EventSchedulerSystemComponent;
+    using RecastNavigation::RecastNavigationMeshRequestBus;
+    using RecastNavigation::RecastNavigationMeshRequests;
+    using RecastNavigation::RecastNavigationDebugDraw;
+    using RecastNavigation::DetourNavigationRequests;
+    using RecastNavigation::NavMeshQuery;
+    using RecastNavigation::DetourNavigationComponent;
+    using RecastNavigation::DetourNavigationRequestBus;
+    using RecastNavigationTests::Wait;
+    using RecastNavigationTests::MockShapeComponent;
+
+    class EditorNavigationTest
+        : public UnitTest::AllocatorsFixture
+    {
+    public:
+        unique_ptr<AZ::SerializeContext> m_sc;
+        unique_ptr<AZ::BehaviorContext> m_bc;
+        unique_ptr<AZStd::vector<AZ::ComponentDescriptor*>> m_descriptors;
+        unique_ptr<AZ::MockTimeSystem> m_timeSystem;
+        unique_ptr<UnitTest::MockSceneInterface> m_mockSceneInterface;
+        unique_ptr<AzPhysics::SceneQueryHit> m_hit;
+        unique_ptr<UnitTest::MockPhysicsShape> m_mockPhysicsShape;
+        unique_ptr<UnitTest::MockSimulatedBody> m_mockSimulatedBody;
+        unique_ptr<AZ::Console> m_console;
+
+        void SetUp() override
+        {
+            UnitTest::AllocatorsFixture::SetUp();
+
+            m_console.reset(aznew AZ::Console());
+            AZ::Interface<AZ::IConsole>::Register(m_console.get());
+
+            // register components involved in testing
+            m_descriptors = AZStd::make_unique<AZStd::vector<AZ::ComponentDescriptor*>>();
+            m_sc = AZStd::make_unique<AZ::SerializeContext>();
+            m_sc->CreateEditContext();
+            m_bc = AZStd::make_unique<AZ::BehaviorContext>();
+
+            RegisterComponent<RecastNavigationMeshComponent>();
+            RegisterComponent<EditorRecastNavigationMeshComponent>();
+
+            RegisterComponent<RecastNavigationPhysXProviderComponent>();
+            RegisterComponent<EditorRecastNavigationPhysXProviderComponent>();
+
+            RegisterComponent<DetourNavigationComponent>();
+            RegisterComponent<EditorDetourNavigationComponent>();
+
+            RegisterComponent<MockShapeComponent>();
+            RegisterComponent<EventSchedulerSystemComponent>();
+            RegisterComponent<RecastNavigationEditorSystemComponent>();
+
+            m_timeSystem = AZStd::make_unique<NiceMock<AZ::MockTimeSystem>>();
+            m_mockSceneInterface = AZStd::make_unique<NiceMock<UnitTest::MockSceneInterface>>();
+            m_hit = AZStd::make_unique<AzPhysics::SceneQueryHit>();
+            m_mockPhysicsShape = AZStd::make_unique<NiceMock<UnitTest::MockPhysicsShape>>();
+            m_mockSimulatedBody = AZStd::make_unique<NiceMock<UnitTest::MockSimulatedBody>>();
+        }
+
+        void TearDown() override
+        {
+            m_mockSimulatedBody = {};
+            m_mockPhysicsShape = {};
+            m_hit = {};
+            m_mockSceneInterface = {};
+            m_timeSystem = {};
```

Comment:
nit: This is a style thing but `.reset()` is arguable clearer here to indicate resources are being freed - not a deal breaker but just something to consider

Reply:
Done